webapp/stores/notification_store.jsx

```diff
@@ -98,11 +99,22 @@ class NotificationStoreClass extends EventEmitter {
                 duration = parseInt(user.notify_props.desktop_duration, 10) * 1000;
             }
 
+            //Play a sound if explicitly set in settings
             const sound = !user.notify_props || user.notify_props.desktop_sound === 'true';
-            Utils.notifyMe(title, body, channel, teamId, duration, !sound);
 
-            if (sound && !UserAgent.isWindowsApp() && !UserAgent.isMacApp()) {
-                Utils.ding();
+            // Notify if you're not looking in the right channel or when
+            // the window itself is not active
+            const activeChannel = ChannelStore.getCurrent();
+            const inFocus = BrowserStore.getGlobalItem('app-focus', true);
+            const notify = activeChannel.id !== channel.id || !inFocus;
+
+            if (notify) {
+                Utils.notifyMe(title, body, channel, teamId, duration, !sound);
+
+                //Don't add extra sounds on native desktop clients
+                if (sound && !UserAgent.isWindowsApp() && !UserAgent.isMacApp()) {
```

Comment:
I never heard the _ding_ from a mobile app before, but I guess it make sense to check against all "native"  apps